proto/substrait/algebra.proto

```diff
@@ -725,6 +726,35 @@ message NestedLoopJoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+// A delimiter join performs duplicate elimination on one side and then
+// pushes the rows with the duplicates eliminated into an arbitrary
+// number of scans on the opposing side to enact the join.  The keys are
+// used to implement the join condition.
+message DelimiterJoinRel {
```

Comment:
While this looks correct to me as far as the join goes, this also needs to have a corresponding set of `DelimScan` nodes that can be referenced from the delim join. Since these can be nested, there needs to be some way of figuring out which `DelimScan` nodes belong to which `DelimJoin`, so likely the delim scan nodes need to have an index and the `DelimJoin` needs to have a list of indexes/references to the delim scans.

Reply:
What's the difference between a DelimScan node and a projection?  It seems like both construct some value for later consumption.  If a DelimScan's output is merely a new field we can provide the field references of interest to the DelimJoin.  I've looked at the explain results for a  number of queries and it is not as simple as a DelimScan directly providing its results directly to the DelimJoin (for instance TPC-H queries number 17 and 21).